pkg/errlist/errlist.go

```diff
@@ -6,35 +6,132 @@ import (
 	"go/token"
 	"io"
 	"path/filepath"
+	"sort"
 	"strings"
+
+	"encr.dev/pkg/errinsrc"
+	"encr.dev/pkg/errinsrc/srcerrors"
 )
 
+// Verbose controls whether the error list prints all errors
+// or just the first two.
+var Verbose = false
+
+// MaxErrorsToPrint is the maximum number of errors to print
+// if Verbose is false
+var MaxErrorsToPrint = 1
+
 type List struct {
-	list scanner.ErrorList
+	List errinsrc.List `json:"list,omitempty"`
 	fset *token.FileSet
 }
 
+var _ errinsrc.ErrorList = (*List)(nil)
+
 func New(fset *token.FileSet) *List {
 	return &List{fset: fset}
 }
 
+// Convert attempts to convert known error types into an error list
+// if it can't it returns nil
+func Convert(err error) *List {
+	switch err := err.(type) {
+	case *List:
+		return err
+	case *errinsrc.ErrInSrc:
+		l := New(nil)
+		l.List = []*errinsrc.ErrInSrc{err}
+		return l
+	case errinsrc.ErrorList:
+		l := New(nil)
+		l.List = err.ErrorList()
+		return l
+	default:
+		return nil
+	}
+}
+
+// Report is a function that allows you to report an error into
+// this list, without having to check for nil.
+//
+// This function only supports error of types:
+//   - *List
+//   - *errinsrc.ErrInSrc
+//   - *scanner.Error
+//
+// If too many errors have been reported it panics
+// with a Bailout value to abort processing.
+// Use HandleBailout to conveniently handle this.
+func (l *List) Report(err error) {
+	if err == nil {
+		return
+	}
+
+	var errToAdd *errinsrc.ErrInSrc
+	switch err := (err).(type) {
+	case *errinsrc.ErrInSrc:
+		// the base type we expect
+		errToAdd = err
+
+	case *scanner.Error:
+		// errors directly from the Go parser
+		errToAdd = srcerrors.GenericGoParserError(err)
+
+	case scanner.ErrorList:
+		for _, e := range err {
+			l.Report(e)
+		}
+		return
+	case *List:
+		// If it's a different list, then merge it in
+		if err != l {
+			for _, e := range err.ErrorList() {
```

Comment:
I wanted to deprecate `Merge` because that also panic's if the filesets are different, which now they might be. `Report` doesn't care in that case.

Added a deprecation notice to Merge